tools/decompile.py

```diff
@@ -52,23 +52,19 @@ def __init__(self, func_root: Path, function_path: Path, args: argparse.Namespac
     def overlay(self) -> str:
         """Attempts to infer the overlay from the asm file path or returns None if one is not found.
         The '_psp' extension will be included if the version is psp"""
-        nonmatchings_dir = (
-            "f_nonmat"
-            if args.version == "saturn"
-            else "psp" if "psp" in args.version else "nonmatchings"
-        )
+        nonmatchings_dir = "f_nonmat" if args.version == "saturn" else "nonmatchings"
         if not self._overlay:
             nonmatching_index = next(
                 (
                     i
                     for i, part in enumerate(self.relpath.parts)
-                    if part == nonmatchings_dir
+                    if nonmatchings_dir in part
                 ),
                 None,
             )
             self._overlay = (
                 self.relpath.parts[nonmatching_index - 1] if nonmatching_index else None
-            )
+            ).replace("_psp", "")
```

Comment:
What is the issue with _psp being included in the overlay name?

Reply:
It's either here or asm_differ. My logic is that there are no two overlays `ric` and `ric_psp`. There's only one `ric` with multiple versions. It has to be fixed here.

Reply:
In the `asm` directory, the `ric` overlay is identified by `ric_psp`, there is no `ric` directory.
By your logic, shouldn't that be fixed?